npm-shrinkwrap.json

```diff
@@ -2,38 +2,1193 @@
   "name": "convict",
   "version": "1.3.0",
   "dependencies": {
+    "abbrev": {
```

Comment:
Keep in mind, npm 3 does a non-deterministic installation. That means diffs
on shrink-wrap will be larger for new deps submitted by a new contributor,
since it's likely my modules will be installed in a different pattern than
those that existed on any existing contributors machines.

I checked the assert shrink-wrap scripts, but they only appear to ensure
shrink-wrap was performed if any updates to package.json occurred.

This is the first time in my 4 years with nide that I have ever actually
seen shrinkwrap used, so I could be doing something wrong as well.

From what I gather, I was supposed to do an install of the new deps without
modifying package.json, then after ensuring tests pass with the new code, I
updated package.json, then reran `npm shrinkwrap`.

I also had to completely wipe 'node_modules`, because I couldn't just
delete the `node_modules/optimist`... since the non-deterministic install
off other modules that have it in its dep tree caused those modules to fail.

I think this means that with npm 3, any time you remove a dependency, you
are more likely to see a large number of shrinkwrap diffs.
On May 2, 2016 7:16 PM, "Peter deHaan" <notifications@github.com> wrote:

In npm-shrinkwrap.json
<https://github.com/mozilla/node-convict/pull/152#discussion_r61825671>:

> @@ -2,38 +2,1193 @@
>    "name": "convict",
>    "version": "1.3.0",
>    "dependencies": {
> +    "abbrev": {

This is a lot of shrinkwrap diffs!
I think there is a shrinkwrap script in the package.json
<https://github.com/mozilla/node-convict/blob/master/package.json#L34>
which you can use to only add the production dependencies.
This might also be a npm@2 vs npm@3 thing, since they handle shrinkwrap
differently. I'm not sure what version of npm @madarche
<https://github.com/madarche> usually uses.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/mozilla/node-convict/pull/152/files/e3f9d27c82d9c431de85fdbcf2eea44b7670aeda#r61825671>